src/lexitools/commands/correspondences.py

```diff
@@ -0,0 +1,180 @@
+            help='select a threshold')
+    parser.add_argument(
+            '--cutoff',
+            action='store',
+            default=2,
+            type=float,
+            help='select a threshold')
+
+
+def run(args):
+
+    ds = get_dataset(args)
+    clts = args.clts.from_config().api
+    bipa = clts.transcriptionsystem_dict['bipa']
+    color = clts.soundclasses_dict['color']
+    asjp = clts.soundclasses_dict['asjp']
+    sca = clts.soundclasses_dict['sca']
+
+
+    columns=('parameter_id', 'concept_name', 'language_id', 'language_name',
+            'value', 'form', 'segments', 'language_glottocode',
+            'concept_concepticon_id', 'language_latitude',
+            'language_longitude')
+    namespace=(('concept_name', 'concept'), ('language_id', 'doculect'),
+            ('segments', 'tokens'), ('language_glottocode', 'glottolog'),
+            ('concept_concepticon_id', 'concepticon'), ('language_latitude',
+                'latitude'), ('language_longitude', 'longitude'), ('cognacy',
+                    'cognacy'), ('cogid_cognateset_id', 'cogid'))
+
+
+    lex = LexStat.from_cldf(ds.cldf_dir.joinpath('cldf-metadata.json'))
+    args.log.info('Loaded the wordlist')
+
+    G = nx.Graph()
+    for idx, tokens, language in lex.iter_rows('tokens', 'doculect'):
+        for sound in [t.split('/')[1] if '/' in t else t for t in tokens]:
+            try:
+                #G.node[sound]['frequency'] += 1
+                G.node[sound]['language'].add(language)
+            except:
+                G.add_node(
+                        sound, 
+                        frequency=1, 
+                        language=set([language]),
+                        name=bipa[sound].name, 
+                        color=color[sound],
+                        asjp=asjp[sound],
+                        sca=sca[sound],
+                        type=bipa[sound].type
+                        )
+                
+    # add the dummy node for gaps, in case we want to use it
+    G.add_node('-',frequency=0, language=set(lex.cols))
+
+    for node, data in G.nodes(data=True):
+        data['language'] = ', '.join(sorted(data['language']))
+
+
+    for lA, lB in progressbar(combinations(lex.cols, r=2)):
+        for idxA, idxB in lex.pairs[lA, lB]:
+            tokensA, tokensB = (
+                    [t.split('/')[1] if '/' in t else t for t in lex[idxA, 'tokens']],
+                    [t.split('/')[1] if '/' in t else t for t in lex[idxB, 'tokens']]
+                    )
+            langA, langB = lex[idxA, 'doculect'], lex[idxB, 'doculect']
+            classesA, classesB = lex[idxA, 'classes'], lex[idxB, 'classes']
+
+            # check for edit dist == 1
+            if edit_dist(
+                    classesA, 
+                    classesB
+                        ) <= args.threshold:
+                pair = Pairwise(tokensA, tokensB)
+                pair.align()
+                almA, almB, sim = pair.alignments[0]
```

Comment:
The alignment here sometimes changes the tokenization, which leads to unknown segments: for example when aligning " i/j u ³¹" (Fuzhou) to " i/j i/j u ⁴²" (Meixian), we get:

```
['j', 'u', '³¹']
['jj', 'u', '⁴²']
```

And "jj" is unknown because we did not see it in the loop above, so we get a keyError when trying to access this node.

Reply:
this was fixed, I had the error as well.

Reply:
forgot to push my fix to beidasinitic. The alignment is not hte optimal solution, it is just one initial test to work with it. so if you pull most recent beidasinitic, you won't see this error